src/literal.h

```diff
@@ -103,11 +120,12 @@ class Literal {
   static Literals makeZero(Type type);
   static Literal makeSingleZero(Type type);
 
-  static Literal makeNullref() { return Literal(Type(Type::nullref)); }
-  static Literal makeFuncref(Name func) { return Literal(func.c_str()); }
-  static Literal makeExnref(std::unique_ptr<ExceptionPackage> exn) {
-    return Literal(std::move(exn));
+  static Literal makeNull(Type type) {
+    assert(type.isRef());
+    return Literal(type);
   }
+  static Literal makeFunc(Name func) { return Literal(func.c_str()); }
+  static Literal makeExn(ExceptionPackage& package) { return Literal(package); }
```

Comment:
First I renamed `makeNullref` to `makeNull` for clarity, and then figured that it'd be good to rename all of them because we are not creating types but values.